core/src/main/java/bio/terra/pearl/core/service/publishing/PortalEnvPublishable.java

```diff
@@ -4,24 +4,32 @@
 import bio.terra.pearl.core.model.Versioned;
 import bio.terra.pearl.core.model.portal.PortalEnvironment;
 import bio.terra.pearl.core.model.publishing.*;
-import bio.terra.pearl.core.model.study.StudyEnvironment;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.Stream;
 
-public interface Publishable {
-    public static final List<String> CONFIG_IGNORE_PROPS = List.of("id", "createdAt", "lastUpdatedAt", "class",
-            "studyEnvironmentId", "portalEnvironmentId", "emailTemplateId", "emailTemplate",
-            "consentFormId", "consentForm", "surveyId", "survey", "versionedEntity", "trigger");
-    public void loadForDiffing(PortalEnvironment portalEnv);
-    public void loadForDiffing(StudyEnvironment studyEnv);
-    public void updateDiff(PortalEnvironment sourceEnv, PortalEnvironment destEnv, PortalEnvironmentChange change);
-    public void updateDiff(StudyEnvironment sourceEnv, StudyEnvironment destEnv, StudyEnvironmentChange change);
-    public void applyDiff(PortalEnvironment sourceEnv, PortalEnvironment destEnv, PortalEnvironmentChange change);
-    public void applyDiff( StudyEnvironmentChange change, StudyEnvironment destEnv, PortalEnvironment destPortalEnv);
+/** For service classes handling object attached to a PortalEnvironment that can be published across environments */
+public interface PortalEnvPublishable {
+    List<String> DEFAULT_PUBLISH_IGNORE_PROPS = List.of("id", "createdAt", "lastUpdatedAt", "class",
+            "portalEnvironmentId", "studyEnvironmentId");
```

Comment:
since this is the portal env interface, does studyEnvironmentId need to be excluded here?

Reply:
ooof, so this was my least favorite part of the PR.  because of the way that inheritance of interfaces in Java works, StudyEnvPublishable can't have a DEFAULT_PUBLISH_IGNORE_PROPS if you want to have the ability to inherit both.  the compiler will throw an error that one is hiding the other.  So I crammed them both into here.